### PR TITLE
Add Holt-Winters exponential smoothing

### PR DESCRIPTION
**Pull request by coffepowered**
## Why are these changes needed?
Add a new model for timeseries: Holt-Winters. It's fast to train and has a small set of hyperparameters

## Checks

- [x] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [x] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

File: flaml/automl/task/generic_task.py [hunk 1]
```diff
@@ -819,9 +819,14 @@ def default_estimator_list(self, estimator_list: List[str]) -> List[str]:
...
-                        estimator_list += ["prophet", "arima", "sarimax"]
+                        estimator_list += [
+                            "prophet",
+                            "arima",
+                            "sarimax",
+                            "holt-winters",
+                        ]
                     except ImportError:
-                        estimator_list += ["arima", "sarimax"]
+                        estimator_list += ["arima", "holt-winters"]
```
Comment (sonichi) [inline flaml/automl/task/generic_task.py:829]: Is there a good reason to remove "sarimax" here?

File: test/automl/test_forecast_hw.py
Comment (sonichi) [inline test/automl/test_forecast_hw.py:1]: Do we need to add a new test file here? Most test cases seem to be already covered in test_forecast.py. Adding a few test cases in existing or new test functions in the same file may work.

File: flaml/automl/model.py [hunk 2]
```diff
@@ -2028,6 +2028,104 @@ def fit(self, X_train, y_train, budget=None, free_mem_ratio=0, **kwargs):
...
+    The class for tuning Holt Winters model, aka 'Triple Exponential Smoothing'.
+    """
+
+    @classmethod
+    def search_space(cls, **params):
+        space = {
+            "damped_trend": {
+                "domain": tune.choice([True, False]),
+                "init_value": False,
+                "low_cost_init_value": False,
```
Comment (sonichi) [inline flaml/automl/model.py:2042]: "low_cost_init_value" is not needed here. "init_value" is enough.

File: flaml/automl/model.py [hunk 3]
```diff
@@ -2028,6 +2028,104 @@ def fit(self, X_train, y_train, budget=None, free_mem_ratio=0, **kwargs):
...
+            },
+            "trend": {
+                "domain": tune.choice(["add", "mul", None]),
+                "init_value": "add",
+                "low_cost_init_value": "add",
```
Comment (sonichi) [inline flaml/automl/model.py:2047]: Remove

File: flaml/automl/model.py [hunk 4]
```diff
@@ -2028,6 +2028,104 @@ def fit(self, X_train, y_train, budget=None, free_mem_ratio=0, **kwargs):
...
+            },
+            "seasonal": {
+                "domain": tune.choice(["add", "mul", None]),
+                "init_value": "add",
+                "low_cost_init_value": None,
```
Comment (sonichi) [inline flaml/automl/model.py:2052]: remove

File: flaml/automl/model.py [hunk 5]
```diff
@@ -2028,6 +2028,104 @@ def fit(self, X_train, y_train, budget=None, free_mem_ratio=0, **kwargs):
...
+            },
+            "use_boxcox": {
+                "domain": tune.choice([False, True]),
+                "init_value": False,
+                "low_cost_init_value": True,
```
Comment (sonichi) [inline flaml/automl/model.py:2057]: remove

File: flaml/automl/model.py [hunk 6]
```diff
@@ -2028,6 +2028,104 @@ def fit(self, X_train, y_train, budget=None, free_mem_ratio=0, **kwargs):
...
+            },
+            "seasonal_periods": {  # statsmodels casts this to None if "seasonal" is None
+                "domain": tune.choice(
+                    [7, 12, 4, 52, 6]
+                ),  # weekly, yearly, quarterly, weekly w yearly data
+                "init_value": 7,
+                "low_cost_init_value": 7,
```
Comment (sonichi) [inline flaml/automl/model.py:2064]: remove

File: flaml/automl/model.py [hunk 7]
```diff
@@ -2028,6 +2028,104 @@ def fit(self, X_train, y_train, budget=None, free_mem_ratio=0, **kwargs):
...
+        current_time = time.time()
+        train_df = self._join(X_train, y_train)
+        train_df = self._preprocess(train_df)
+        regressors = list(train_df)
+        regressors.remove(TS_VALUE_COL)
+        if regressors:
+            # Not supported ATM. A few discussions here:
+            # https://robjhyndman.com/hyndsight/ets-regressors/
+            # https://stats.stackexchange.com/questions/220830/holt-winters-with-exogenous-regressors-in-r
+            logger.debug("Regressors are ignored for Holt-Winters ETS models.")
```
Comment (sonichi) [inline flaml/automl/model.py:2086]: What about
```suggestion
            logger.warning("Regressors are ignored for Holt-Winters ETS models.")
```

File: flaml/automl/task/generic_task.py (hunk 1 above)
Comment (coffepowered) [reply flaml/automl/task/generic_task.py]: This was a mistake, corrected

File: test/automl/test_forecast_hw.py
Comment (coffepowered) [reply test/automl/test_forecast_hw.py:1]: test_forecast_hw is one of the reasons why this is still marked as a draft. Did not expect a review so fast or while in draft, but well received! File removed :)

File: flaml/automl/model.py (hunk 2 above)
Comment (coffepowered) [reply flaml/automl/model.py]: good point, this model is very cheap to fit. Thank you

File: flaml/automl/model.py (hunk 7 above)
Comment (coffepowered) [reply flaml/automl/model.py:2072]: I'm happy to accept the suggestion, but is there a place where we can document this kind of choice? I'd personally find it helpful. Just a humble proposal, you may have good reasons for not being willing to maintain such kind of documentation

Comment (sonichi) [reply flaml/automl/model.py]: Thanks for this proposal. We should consider adding such documentation. Perhaps under https://microsoft.github.io/FLAML/docs/Contribute or a subpage.

**Review by EgorKraevTransferwise — APPROVED** [review]

**Review by sonichi — APPROVED** [review]

**Pull request merged**